.github/workflows/CI-rustdoc.yml

```diff
@@ -0,0 +1,71 @@
+name: CI-rustdoc
+
+run-name: "Workflow performing CI step: Rust doc generation"
+
+on:
+  workflow_call:
+  workflow_dispatch:
+
+env:
+  RUST_BACKTRACE: 1
+
+jobs:
+  rustdoc-generation:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout branch
+        uses: actions/checkout@v4
+
+      - name: Set up cargo cache
+        uses: actions/cache@v4
+        continue-on-error: false
+        with:
+          path: |
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            target/
+          key: ${{ runner.os }}-lint-format-cargo-${{ hashFiles('**/Cargo.lock') }}
```

Comment:
Is `lint-format` cache here used on purpose? My feeling would be to use `build-test` cache, but if your is a motivated choice just ignore this comment.

Reply:
It makes sense to me, thanks! I've also updated the `cargo doc` command adding the `--release` flag as the cache is built with the same flag.